content/recipes/interoperability/registering-swisslipids-identifiers.md

```diff
@@ -0,0 +1,162 @@
+
+This script uses a federated query against https://beta.sparql.swisslipids.org/sparql/ after a suggestion by Dr [Jerven Bolleman](https://orcid.org/0000-0002-7449-1266)
+who indicated that the [RDF4J](https://rdf4j.org/) backing this SPARQL endpoint will automatically batch the query against Wikidata, overcoming
+limitations of the Wikidata Query Service:
+
+```sparql
+PREFIX wdt: <http://www.wikidata.org/prop/direct/>
+SELECT ?wd ?key ?value WHERE {
+  SERVICE <https://query.wikidata.org/sparql> {
+    SELECT (substr(str(?compound),32) as ?wd) ?key ?value WHERE {
+      ?compound wdt:P235 ?key .
+      OPTIONAL { ?compound wdt:P8691 ?value . }
+    }
+  }
+}
+```
+
+This creates a file that looks like:
+
+```shell
+Q76738581       P8691   "SLM:000163948" S248    Q41165322       S854    "https://www.swisslipids.org/#/downloads"       S813    +2021-11-06T00:00:00Z/11
+Q76865359       P8691   "SLM:000163954" S248    Q41165322       S854    "https://www.swisslipids.org/#/downloads"       S813    +2021-11-06T00:00:00Z/11
+Q76865370       P8691   "SLM:000163964" S248    Q41165322       S854    "https://www.swisslipids.org/#/downloads"       S813    +2021-11-06T00:00:00Z/11
+Q76866423       P8691   "SLM:000163966" S248    Q41165322       S854    "https://www.swisslipids.org/#/downloads"       S813    +2021-11-06T00:00:00Z/11
+Q76865004       P8691   "SLM:000163968" S248    Q41165322       S854    "https://www.swisslipids.org/#/downloads"       S813    +2021-11-06T00:00:00Z/11
+Q76733356       P8691   "SLM:000164019" S248    Q41165322       S854    "https://www.swisslipids.org/#/downloads"       S813    +2021-11-06T00:00:00Z/11
+Q76733312       P8691   "SLM:000164023" S248    Q41165322       S854    "https://www.swisslipids.org/#/downloads"       S813    +2021-11-06T00:00:00Z/11
+Q76737210       P8691   "SLM:000164026" S248    Q41165322       S854    "https://www.swisslipids.org/#/downloads"       S813    +2021-11-06T00:00:00Z/11
+Q76736881       P8691   "SLM:000164032" S248    Q41165322       S854    "https://www.swisslipids.org/#/downloads"       S813    +2021-11-06T00:00:00Z/11
+Q76735022       P8691   "SLM:000164034" S248    Q41165322       S854    "https://www.swisslipids.org/#/downloads"       S813    +2021-11-06T00:00:00Z/11
+```
+
+This resulted in about 17.5 thousand mappings. These are based on exact InChIKey match. 
+
+### Step 5: running the QuickStatements
+
+The resulting mappings are being added via the [QuickStatements website](https://quickstatements.toolforge.org/).
+
+## Conclusion
+
+It will be clear Bacting can be replaced without too much effort by other tools making the same steps. What is the
+main conclusion is that we can indeed automate adding external identifiers routinely to Wikidata using InChIKey
+matching.
+
+What this protocol does not do, however, is create Wikidata items for lipids in the SwissLipids database that
+are not yet in Wikidata. This is the topic of a future recipe.
+
```

Comment:
```suggestion
The main conclusion is that we can automate adding external identifiers routinely to Wikidata using InChIKey matching. 

It will be clear [Bacting tool](https://bio.tools/bacting) can be replaced without too much effort by other tools making the same steps.

What this protocol does not do, however, is create Wikidata items for lipids in the SwissLipids database that
are not yet in Wikidata. 

This is the topic of a future recipe.

```